pkg/cluster/types.go

```diff
@@ -22,10 +22,10 @@ import (
 // Upgrader is the public interface to the upgrade module used by the plugin.
 type Upgrader interface {
 	Initialize(ctx context.Context, cs *api.OpenShiftManagedCluster) error
-	Deploy(ctx context.Context, cs *api.OpenShiftManagedCluster, azuretemplate map[string]interface{}, deployFn api.DeployFn) error
```

Comment:
the fact that Initialize() above is not changed implies to me that it should not be part of this interface.  Remove it from the interface, and convert its implementation to `initialize()` (unexported).